mmdet/apis/ote/apis/detection/inference_task.py

```diff
@@ -213,7 +214,7 @@ def _add_predictions_to_dataset(self, prediction_results, dataset, confidence_th
                                 box_points = cv2.boxPoints(cv2.minAreaRect(contour))
                                 points = [Point(x=point[0] / width, y=point[1] / height) for point in box_points]
                             labels = [ScoredLabel(self._labels[label_idx], probability=probability)]
-                            shapes.append(Annotation(Polygon(points=points), labels=labels, id=label_idx))
+                            shapes.append(Annotation(Polygon(points=points), labels=labels, id=ID(f"{label_idx:08}")))
```

Comment:
What's the idea behind this formatting? Is it enough to cast the value to string or not?

Reply:
Such formatting is required for correct sorting of labels, since the string "10" will be considered less than the string "2"

Reply:
Could you please point me to place where sorted order matters?

Reply:
No, I can't, a similar request came with the same changes for mmsegmentation

Reply:
If that is enough to change such lines to "ID(label_idx)", without string formatting, then I can make this change

Reply:
I think we need to have an unified decision for all algo backends for all types of id-s.
When we made just convertion to str, in mmsegmentation there was an issue with wrong elements order in Datasets due to `"1" < "10" < "2"`
Due to it we made the current decision `ID(f"{label_idx:08}")`.
It could also be an issue even in mmdetection if somebody creates elements one-by-one, and then sort elements, the order of elements will be wrong.

So, I think the current decision `ID(f"{label_idx:08}")` is ok and I will insist on this decision.

Reply:
1. If you need a unified decision then SDK is the right place to introduce it.
2. Changes like this are likely hiding the issue (if it really exists), rather than solving and addressing it.
3. This being said, from the algo-backend itself the change looks harmless, so I can accept it.